common/functions/src/scalars/hashes/hash.rs

```diff
@@ -12,28 +12,59 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use crate::scalars::function_factory::FunctionFactory;
+use std::collections::hash_map::DefaultHasher;
+
+use twox_hash::XxHash32;
+use twox_hash::XxHash64;
+
+use super::BaseHashFunction;
 use crate::scalars::Blake3HashFunction;
+use crate::scalars::Function2Factory;
 use crate::scalars::Md5HashFunction;
 use crate::scalars::Sha1HashFunction;
 use crate::scalars::Sha2HashFunction;
-use crate::scalars::SipHashFunction;
-use crate::scalars::XxHash32Function;
-use crate::scalars::XxHash64Function;
 
 #[derive(Clone)]
 pub struct HashesFunction;
 
+pub type XxHash32Function = BaseHashFunction<XxHash32, u32>;
+pub type XxHash64Function = BaseHashFunction<XxHash64, u64>;
+pub type SipHash64Function = BaseHashFunction<DefaultHasher, u64>;
+
 impl HashesFunction {
-    pub fn register(factory: &mut FunctionFactory) {
-        factory.register("siphash", SipHashFunction::desc());
-        factory.register("siphash64", SipHashFunction::desc());
+    pub fn register2(factory: &mut Function2Factory) {
         factory.register("md5", Md5HashFunction::desc());
         factory.register("sha", Sha1HashFunction::desc());
         factory.register("sha1", Sha1HashFunction::desc());
         factory.register("sha2", Sha2HashFunction::desc());
+
         factory.register("blake3", Blake3HashFunction::desc());
         factory.register("xxhash32", XxHash32Function::desc());
         factory.register("xxhash64", XxHash64Function::desc());
+        factory.register("siphash64", SipHash64Function::desc());
+        factory.register("siphash", SipHash64Function::desc());
+    }
+}
+
+#[cfg(test)]
```

Comment:
It needs to be placed in the right place

Reply:
I noticed that there are some other files that contain similar tests mod. I think they may not be involved in the compilation (with `test = false` in Cargo.toml).